lib/cache.sh

```diff
@@ -48,6 +48,7 @@ restore_default_cache_directories() {
   local build_dir=${1:-}
   local cache_dir=${2:-}
   local yarn_cache_dir=${3:-}
+  local npm_cache="$(npm config get cache)"
```

Comment:
We set `NPM_CONFIG_CACHE` to a `/tmp` dir in `bin/compile`. Reading that is faster and unlikely to error compared to shelling out. Maybe we should use that instead?

Reply:
Since there's a pattern of passing `yarn_cache_dir` in, maybe we should do that for the `npm` version too? I think your a fan of functions with fewer side-effects if I recall =).

Reply:
yes, that cleans things up nicely.  i completely missed the `NPM_CONFIG_CACHE` variable until you pointed it out